ecommerce/api_test.py

```diff
@@ -357,6 +357,54 @@ def test_get_valid_coupon_versions_after_redemption(user, is_global):
     assert list(get_valid_coupon_versions(product, user)) == []
 
 
+@pytest.mark.parametrize("is_global", [True, False])
+def test_get_valid_coupon_versions_with_max_redemptions_per_user(user, is_global):
+    """
+    Verify that the correct valid CouponPaymentVersions are returned before and after redemption
+    """
+    AMOUNT = "1.00000"
+    COUPON_CODE = "TESTCOUPON1"
+    COUPON_TYPE = "promo"
+    MAX_REDEMPTIONS = 3  # Must be greater or equal to max_redemptions_per_user
+    MAX_REDEMPTIONS_PER_USER = 3
+
+    payment = CouponPaymentFactory()
+    cpv = CouponPaymentVersionFactory(
+        payment=payment,
+        amount=Decimal(AMOUNT),
+        coupon_type=COUPON_TYPE,
+        max_redemptions=MAX_REDEMPTIONS,
+        max_redemptions_per_user=MAX_REDEMPTIONS_PER_USER,
+        num_coupon_codes=1,
+    )
+    coupon = CouponFactory(
+        payment=payment, coupon_code=COUPON_CODE, is_global=is_global
+    )
+    cv = CouponVersionFactory(payment_version=cpv, coupon=coupon)
+    expected = [cv]
```

Comment:
```suggestion
    expected_coupon_versions = [cv]
```

Reply:
Also, change it wherever it's being used.

Reply:
Updated.